packages/bruno-common/src/interpolate/index.ts

```diff
@@ -13,19 +13,44 @@
 
 import { flattenObject } from '../utils';
 
+function serializeObject(obj: Object) {
+  // Check if the object has a `toString` method like `Moment`
+  // Don't do it with arrays they serialize weirdly
+  if (typeof obj.toString === 'function' && !Array.isArray(obj)) {
```

Comment:
I think this can lead to really misleading behavior. Consider the case where an object has a custom `toString` method: It would be called, but only, if the object is at the top level. If it is nested, then the default `JSON.stringify()` method is used, which leads to a different result.

Example: 
```js
// object on top-level
const today = moment();
console.log(today.toString());
// -> Wed Mar 27 2024 09:43:06 GMT+0100

// object nested:
const nestedExample = {"nestedExample": today}
console.log(JSON.stringify(nestedExample))
// -> {"nestedExample":"2024-03-27T08:43:06.249Z"}
```

See [https://jsfiddle.net/17Lzbjwm/1/](https://jsfiddle.net/17Lzbjwm/1/)

Proposal for simpler logic for `{{thing}}` replacement:
- if `thing` is a string, do a literal replacement
- if `thing´ is anything else, use `JSON.stringify(thing)`
  - in case the user does not like the result, they have to do it themselves in prerequest or postrequest scripts.

Reply:
I see, the whole point of calling `toString` was, because `JSON.stringify` would add double quotes around strings. I'll look into it later the day.